org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend

```diff
@@ -3308,8 +3547,9 @@ class WorkspaceEdit {
 	 * version of a text document. Whether a client supports versioned document
 	 * edits is expressed via `WorkspaceClientCapabilities.versionedWorkspaceEdit`.
 	 */
-	List<TextDocumentEdit> documentChanges
-	
+	@JsonAdapter(DocumentChangeListAdapter)
+	List<Either<TextDocumentEdit, ResourceOperation>> documentChanges
```

Comment:
If I understand correctly, this protocol change replaces the previous `@Beta` status `resourceChanges` property introduced in #163. Please mark the `resourceChanges` property and the `ResourceChange` class as `@Deprecated`, pointing to the new API with the `@deprecated` JavaDoc tag.